subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/ReportImplementation.kt

```diff
@@ -308,16 +298,21 @@ class ReportImplementation(
 
     private fun addEntry(entry: Entry) {
         val started = getCastedStateOrNull<ReportState.Initialized.Started>()
-        val entriesList = started?.getCurrentStepOrCreate {
+        val entriesList = started
+            ?.currentStepOrSynthetic()
+            ?.entryList ?: earlyEntries
+        entriesList.add(entry)
+    }
+
+    private fun ReportState.Initialized.Started.currentStepOrSynthetic() =
+        getCurrentStepOrCreate {
             StepResult(
                 isSynthetic = true,
                 timestamp = timeProvider.nowInSeconds(),
-                number = started.stepNumber++,
-                title = "Synthetic step"
+                number = stepNumber++,
+                title = "Out of step"
```

Comment:
@Kutashov asked me to rename it because it's not clear for users what is `Synthetic`

Reply:
It might stay unchanged, but I would request some comprehensive on-hover description, if it's possible.

Reply:
Could you propose something clear for you

Reply:
I will think about that